codex/stores/networkstore.nim

```diff
@@ -31,28 +31,27 @@ type
     engine*: BlockExcEngine # blockexc decision engine
     localStore*: BlockStore # local block store
 
-method getBlock*(self: NetworkStore, cid: Cid): Future[?! (? bt.Block)] {.async.} =
+method getBlock*(self: NetworkStore, cid: Cid): Future[?!bt.Block] {.async.} =
   ## Get a block from a remote peer
   ##
 
-  trace "Getting block from network store", cid
-
-  let blk = await self.localStore.getBlock(cid)
-  if blk.isErr:
-    return blk
-  if blk.get.isSome:
-    trace "Retrieved block from local store", cid
-    return blk
-
-  trace "Block not found in local store", cid
-  try:
-    # TODO: What if block isn't available in the engine too?
-    let blk = await self.engine.requestBlock(cid)
-    # TODO: add block to the local store
-    return blk.some.success
-  except CatchableError as exc:
-    trace "Exception requesting block", cid, exc = exc.msg
-    return failure(exc)
+  trace "Getting block from local store or network", cid
+
+  let
+    getRes = await self.localStore.getBlock(cid)
+
+  if getRes.isErr:
+    if not (getRes.error of BlockNotFoundError): return getRes
+    trace "Block not in local store", cid
+    try:
+      # TODO: What if block isn't available in the engine too?
+      return ok await self.engine.requestBlock(cid)
```

Comment:
you can probably do an `(await self.engine.requestBlock(cid)).catch`, which would convert to a questionable result.

Reply:
I reworked it using `.catch` and `without`.